### DISCUSSION TITLE
Reading text files from a folder within app project

### DISCUSSION THREAD

**Discussion by nubilion** [zepp-health/zeppos-forum#48] [Q&A]
I would like to read a text file which is stored in a specific project directory, e.g:

/assets/

or

/data/

Are we able to do this by using hmFS File system module?

If not, which is the preferred way of reading text files packed within the application?
Thank you in advance!

**Comment by htoooth** [comment]
Hi @nubilion:
Yes, you can read file from assets directory. But data is runtime directory.

This [link](https://docs.zepp.com/docs/1.0/guides/framework/device/fs/?_highlight=raw#assets-resource-directory) is useful for you.

**Reply by MyoungXUE** [reply]
As a side note, if you want to put it in the `/assets` directory, it is recommended to put the text file in the `/assets/raw` directory
